src/routes/mayan/MayanRouteBase.ts

```diff
@@ -140,24 +150,38 @@ export class MayanRouteBase<N extends Network> extends routes.AutomaticRoute<
     return ['Mainnet', 'Testnet'];
   }
 
-  static supportedChains(network: Network): Chain[] {
-    return supportedChains(network);
-  }
-
   static isProtocolSupported<N extends Network>(
     chain: ChainContext<N>,
   ): boolean {
     return supportedChains(chain.network).includes(chain.chain);
   }
 
+  /**
+   * Filter a list of tokens to only include USDC tokens
+   */
+  private static filterUSDCTokens(chain: Chain, tokens: TokenId[]): TokenId[] {
+    return tokens.filter((tk) =>
+      isCanonicalUSDCToken(chain, tk.address.toString()),
+    );
+  }
+
   // Mayan can handle any input and output token that has liquidity on a DeX
   static async supportedDestinationTokens<N extends Network>(
     _token: TokenId,
-    _fromChain: ChainContext<N>,
+    fromChain: ChainContext<N>,
     toChain: ChainContext<N>,
   ): Promise<TokenId[]> {
-    // TODO update this function to only allow HyperCore/USDC
-    return getAllTokenIdsForChain(toChain.chain);
+    const tokens = getAllTokenIdsForChain(toChain.chain);
+
+    // For HyperCore, only allow USDC as destination token
+    if (isHyperCoreChain(toChain.chain)) {
+      if (!isEvmChain(fromChain.chain)) {
+        return [];
```

Comment:
This particular check may not be needed if the destination chain gets filtered out, since the chain won't show up in the list. But double check.